crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1124,16 +1135,6 @@ impl<'db> Type<'db> {
 
             (Type::Dynamic(_), Type::Dynamic(_)) => true,
 
-            (Type::Instance(instance), Type::SubclassOf(subclass))
-            | (Type::SubclassOf(subclass), Type::Instance(instance)) => {
-                let Some(base_class) = subclass.subclass_of().into_class() else {
-                    return false;
-                };
-
-                instance.class.is_known(db, KnownClass::Type)
-                    && base_class.is_known(db, KnownClass::Object)
-            }
-
```

Comment:
I noticed that this branch was redundant, since we now eagerly normalize `type[object]` to `type`. But I could separate this out into another PR, if we prefer that